contracts/storage/storage.hpp

```diff
@@ -0,0 +1,170 @@
+#include <eoslib/eos.hpp>
+#include <eoslib/token.hpp>
+#include <eoslib/db.hpp>
+
+/**
+ *  @defgroup examplecontract Example Storage
+ *  @brief Example Contract
+ *  @ingroup contractapi
+ *
+ */
+
+/**
+ * Make it easy to change the account name the storage is deployed to.
+ */
+#ifndef TOKEN_NAME
+#define TOKEN_NAME TOK
+#endif
+
+namespace TOKEN_NAME {
+
+    /**
+    * @defgroup storageapi Storage Contract
+    * @brief Defines the storage contract example
+    * @ingroup examplecontract
+    *
+    * @{
+    */
+
+    /**
+    * Defines a storage token
+    */
+    typedef eos::token<uint64_t, N(storage)> StorageTokens;
+
+   /**
+    * Transfer requires that the sender and receiver be the first two 
+    * accounts notified and that the sender has provided authorization.
+    */
+
+   struct Transfer {
+        /**
+        * Account to transfer from
+        */
+        AccountName from;
+
+        /**
+        * Account to transfer to
+        */
+        AccountName to;
+
+        /**
+        * quantity to transfer
+        */
+        StorageTokens quantity; 
```

Comment:
We don't create StorageTokens and storage, instead we have storage quota and storage consumed and its measured in bytes.